In [7]:
import json
import pandas as pd
import re

# JSON 파일 경로 설정
json_file_path = '2024.06/youtube_videos_2024-06-11.json'
csv_file_path = 'youtube_data/youtube_videos_2024-06-11.csv'

# JSON 파일 읽기
with open(json_file_path, 'r', encoding='utf-8') as file:
    data = json.load(file)

# 한자가 포함된 행을 제외하는 함수 정의
def contains_foreign_language(text):
    # 정규표현식: 영문자, 숫자, 공백, 기본 특수문자를 제외한 다른 문자
    return bool(re.search(r'[^\x00-\x7F]', text))

# 데이터 필터링
filtered_data = [line for line in data if not contains_foreign_language(line)]

# 데이터프레임으로 변환
df = pd.DataFrame(filtered_data, columns=['title'])

# CSV 파일로 저장
df.to_csv(csv_file_path, index=False)

print(f"Filtered data saved to {csv_file_path}")

Filtered data saved to youtube_data/youtube_videos_2024-06-11.csv


In [11]:
import json
import pandas as pd
import re
import os

# 한자가 포함된 행을 제외하는 함수 정의
def contains_chinese(text):
    return bool(re.search(r'[\u4e00-\u9fff]', text))

# 날짜 범위 설정
start_date = 1
end_date = 30

# 기본 경로 설정
json_base_path = '2024.06'
csv_base_path = 'youtube_data'

# 폴더가 없는 경우 생성
if not os.path.exists(csv_base_path):
    os.makedirs(csv_base_path)

# 각 날짜에 대해 파일 처리
for day in range(start_date, end_date + 1):
    json_file_path = f'{json_base_path}/youtube_videos_2024-06-{day:02d}.json'
    csv_file_path = f'{csv_base_path}/youtube_videos_2024-06-{day:02d}.csv'

    # JSON 파일 읽기
    try:
        with open(json_file_path, 'r', encoding='utf-8') as file:
            data = json.load(file)

        # 데이터 필터링
        filtered_data = [line for line in data if not contains_foreign_language(line)]

        # 데이터프레임으로 변환
        df = pd.DataFrame(filtered_data, columns=['title'])

        # CSV 파일로 저장
        df.to_csv(csv_file_path, index=False)

        print(f"Filtered data saved to {csv_file_path}")

    except FileNotFoundError:
        print(f"File not found: {json_file_path}")
    except json.JSONDecodeError:
        print(f"Error decoding JSON from file: {json_file_path}")
    except Exception as e:
        print(f"An error occurred while processing {json_file_path}: {e}")

Filtered data saved to youtube_data/youtube_videos_2024-06-01.csv
Filtered data saved to youtube_data/youtube_videos_2024-06-02.csv
Filtered data saved to youtube_data/youtube_videos_2024-06-03.csv
Filtered data saved to youtube_data/youtube_videos_2024-06-04.csv
Filtered data saved to youtube_data/youtube_videos_2024-06-05.csv
Filtered data saved to youtube_data/youtube_videos_2024-06-06.csv
Filtered data saved to youtube_data/youtube_videos_2024-06-07.csv
Filtered data saved to youtube_data/youtube_videos_2024-06-08.csv
Filtered data saved to youtube_data/youtube_videos_2024-06-09.csv
Filtered data saved to youtube_data/youtube_videos_2024-06-10.csv
Filtered data saved to youtube_data/youtube_videos_2024-06-11.csv
Filtered data saved to youtube_data/youtube_videos_2024-06-12.csv
Filtered data saved to youtube_data/youtube_videos_2024-06-13.csv
Filtered data saved to youtube_data/youtube_videos_2024-06-14.csv
Filtered data saved to youtube_data/youtube_videos_2024-06-15.csv
Filtered d

# 입력 데이터 만들기

In [12]:
df = pd.read_csv("youtube_data/youtube_videos_2024-06-01.csv")

In [13]:
df

title
0    This Is How You Can Trade The Nvidia Stock Split
1   NVDA Nvidia Red or Green on Monday...$1050 or ...
2   Should I Buy NVDA? | Weekend Edition June 1, 2024
3   Is NVIDIA Stock A BUY Before Stock Split For M...
4   NVDA Stock: (NVIDIA stock) NVDA STOCK Predicti...
5                  Nvidia Stock TRAP: Beware Of This!
6              NVDA Stock: JUNE CRAZY! (NVIDIA stock)
7   STILL EARLY! Why I&#39;m STILL Buying Nvidia S...
8   NVDA Nvidia Price Range Analysis ... $1200 or ...
9   What No One Tells You About NVDA 10 to 1 Stock...
10  Billionaires Are Offloading Nvidia and Investi...
11          NVDA stock - Is Nvidia Stock (NVDA) a Buy
12   Using Long Call Options to Trade Nvidia Earnings
13  I Can&#39;t Stay Quiet on Nvidia Stock (NVDA) ...
14  NVDA Stock: (NVIDIA stock) NVDA STOCK Predicti...
15  Nvidia Stock Analysis | $1300 Or $975 Before S...
16                                Great Input on NVDA
17          NVDA peek at the monthly &amp; the daily!
18      Buy These Stocks in 2024 And Never Work Again
19  Is Vanguard&#39;s VTEB the Ultimate Tax-Free I...
20   $NVDA #Nvidia Cycles? Whale data? Price targets?
21  CLOSING STREAM! Stock Market Live! SPY. QQQ. G...
22  Cisco&#39;s Q3 2024 Earnings Review #Cisco #Wa...
23  NVIDIA STOCK IS GOING TO SPLIT #nvidia #stocks...
24  The Absurdity of Nvidia&#39;s Profits #nvidia ...
25  They don&#39;t care about you #gamestopstock #...
26                WATCH THIS, NVDA BIG MANIPULATION ,
27  New Traders Vs Good Traders #tradingview | Sto...
28                Session Record in Live 12 with NVDA
29     NVIDIA EARNINGS RECAP #trading #stocks #nvidia
30  TECH STOCK ANALYSIS (Amazon, Google, Meta, App...
31  See how AI is boosting internal efficiency and...
32                             Market Update  5/31/24
33  [LIVE TRADING} HOW TO PLAN -TRADE - RECAP : SH...
34  Why NVIDIA Data Center Dominates Amid AI Boom ...
35  How to Trade #coin #aapl #nvda #amd   VIP-ZEN ...
36  E7 NVIDIA AI BUBBLE  - We Can&#39;t Stay Quiet...

0601
- "This Is How You Can Trade The Nvidia Stock Split": 70점
- "NVDA Nvidia Red or Green on Monday...$1050 or $1130": 65점
- "Should I Buy NVDA? | Weekend Edition June 1, 2024": 75점
- "Is NVIDIA Stock A BUY Before Stock Split For MASSIVE Gains?! | NVDA Stock Analysis! |": 85점
- "NVDA Stock: (NVIDIA stock) NVDA STOCK Prediction NVDA STOCK Analysis NVDA STOCK NEWS TODAY $NVDA": 80점
- "Nvidia Stock TRAP: Beware Of This!": 40점
- "NVDA Stock: JUNE CRAZY! (NVIDIA stock)": 60점
- "STILL EARLY! Why I'm STILL Buying Nvidia Stock NVDA": 95점
- "NVDA Nvidia Price Range Analysis ... $1200 or $1400?": 50점
- "What No One Tells You About NVDA 10 to 1 Stock Split": 55점
- "Billionaires Are Offloading Nvidia and Investing in This Stock Instead | NASDAQ: NVDA | Stock Market": 45점
- "NVDA stock - Is Nvidia Stock (NVDA) a Buy": 72점
- "Using Long Call Options to Trade Nvidia Earnings": 78점
- "I Can't Stay Quiet on Nvidia Stock (NVDA) Crashing Any More": 30점
- "NVDA Stock: (NVIDIA stock) NVDA STOCK Prediction NVDA STOCK Analysis NVDA STOCK NEWS TODAY $NVDA": 80점
- "Nvidia Stock Analysis | $1300 Or $975 Before Stock Split? Nvidia Stock Price Prediction": 62점
- "Great Input on NVDA": 88점
- "NVDA peek at the monthly & the daily!": 82점
- "Buy These Stocks in 2024 And Never Work Again": 98점
- "Is Vanguard's VTEB the Ultimate Tax-Free Investment? ETF Explained. #vtebetf": 63점


# 네이버...

In [ ]:
# -*- coding: utf-8 -*-

import base64
import hashlib
import hmac
import requests
import time


class CreateTaskExecutor:
    def __init__(self, host, uri, method, iam_access_key, secret_key, request_id):
        self._host = host
        self._uri = uri
        self._method = method
        self._api_gw_time = str(int(time.time() * 1000))
        self._iam_access_key = iam_access_key
        self._secret_key = secret_key
        self._request_id = request_id

    def _make_signature(self):
        secret_key = bytes(self._secret_key, 'UTF-8')
        message = self._method + " " + self._uri + "\n" + self._api_gw_time + "\n" + self._iam_access_key
        message = bytes(message, 'UTF-8')
        signing_key = base64.b64encode(hmac.new(secret_key, message, digestmod=hashlib.sha256).digest())
        return signing_key

    def _send_request(self, create_request):

        headers = {
            'X-NCP-APIGW-TIMESTAMP': self._api_gw_time,
            'X-NCP-IAM-ACCESS-KEY': self._iam_access_key,
            'X-NCP-APIGW-SIGNATURE-V2': self._make_signature(),
            'X-NCP-CLOVASTUDIO-REQUEST-ID': self._request_id
        }
        result = requests.post(self._host + self._uri, json=create_request, headers=headers).json()
        return result

    def execute(self, create_request):
        res = self._send_request(create_request)
        if 'status' in res and res['status']['code'] == '20000':
            return res['result']
        else:
            return res


if __name__ == '__main__':
    completion_executor = CreateTaskExecutor(
        host='https://clovastudio.apigw.ntruss.com',
        uri='/tuning/v2/tasks',
        method='POST',
        iam_access_key='ncp_iam_BPAMKR3jQRxxYDUIVkn3',
        secret_key='ncp_iam_BPKMKRHKQSqRCWpkP04jWalXK8NZFUvfmF',
        request_id='jihee0250'
    )

    request_data = {'name': 'generation_task',
                    'model': 'HCX-003',
                    'tuningType': 'PEFT',
                    'taskType': 'GENERATION',
                    'trainEpochs': '8',
                    'learningRate': '1e-5f',
                    'trainingDatasetBucket': 'bucket_name',
                    'trainingDatasetFilePath': 'root_path/sub_path/file_name',
                    'trainingDatasetAccessKey': 'access_key',
                    'trainingDatasetSecretKey': 'secret_key'
                    }
    response_text = completion_executor.execute(request_data)
    print(request_data)
    print(response_text)
